src/Microsoft.VisualBasic/tests/VB/TextFieldParserTests.vb

```diff
@@ -0,0 +1,180 @@
+                    Assert.Equal("1", currentRow(0))
+                    Assert.Equal("Test", currentRow(1))
+                    Assert.Equal("99", currentRow(2))
+                    Assert.Equal("Two Words", currentRow(3))
+                    Assert.Equal("Rumnoman", currentRow(4))
+                    Assert.Equal("KS", currentRow(5))
+                    Assert.Equal("80586", currentRow(6))
+                    Assert.Equal("$8333.85", currentRow(7))
+                    Assert.Equal("GREEN", currentRow(8))
+                    Assert.Equal("09/28/1929", currentRow(9))
+                    Assert.Equal("Ipmadaw udoteac vu ote ozehop ane lujat, lar buh cugi lef owici tat liz cogde.
+Fatigiug kojitfu taapi iz alujok me zimipa koz latkekij vem fo si cepzizhub li por ejcirlu.
+Ewujajlu mivec tooju ad bu cowic irtafit ehaoca vojhehfo aztidun zo wezecmo abe muz wikhutwen idce.
+Vesu sejawga tef lahi dirueg si uwmac bidiw nowidza daime sapmim ki casdun urokir tawdac rahaw beiweed.", currentRow(10))
+                    Assert.True(MyReader.EndOfData)
+                End Using
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub FixedFormatTest()
+            Dim TestFile As String = ""
+            Using TestBase As New TextFieldParserTests
+                TestFile = CreateTestFile(TestBase, TestData:=String.Join(vbCrLf, FixedFormatData).ToCharArray, PathFromBase:="", TestFileName:="")
+                Using MyReader As New TextFieldParser(TestFile)
+                    MyReader.TextFieldType = FieldType.FixedWidth
+                    MyReader.SetFieldWidths(FixedFieldWidths)
+
+                    Dim CurrentRow As String()
+                    Dim CurrentRowIndex As Integer = 0
+                    While Not MyReader.EndOfData
+                        CurrentRow = MyReader.ReadFields()
+                        Assert.Equal(FixedFieldWidths.Length, CurrentRow.Length)
+                        Assert.Equal(CurrentRow(0), FixedFormatData(CurrentRowIndex).Substring(0, FixedFieldWidths(0)).Trim)
+                        Assert.Equal(CurrentRow(1), FixedFormatData(CurrentRowIndex).Substring(FixedFieldWidths(0), FixedFieldWidths(1)).Trim)
+                        Assert.Equal(CurrentRow(2), FixedFormatData(CurrentRowIndex).Substring(FixedFieldWidths(0) + FixedFieldWidths(1), FixedFieldWidths(2)).Trim)
+                        Assert.Equal(CurrentRow(3), FixedFormatData(CurrentRowIndex).Substring(FixedFieldWidths(0) + FixedFieldWidths(1) + FixedFieldWidths(2)).Trim)
+                        CurrentRowIndex += 1
+                    End While
+                End Using
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub MultiFormatTest()
+            Dim TestFile As String = ""
+            Using TestBase As New TextFieldParserTests
+                TestFile = CreateTestFile(TestBase, TestData:=String.Join(vbCrLf, FixedFormatData).ToCharArray, PathFromBase:="", TestFileName:="")
```

Comment:
`MultipleFormatData`?

Reply:
This was a test from MSDN it is a file with multiple fixed width formats intermixed in the file

Reply:
> This was a test from MSDN it is a file with multiple fixed width formats intermixed in the file

Which MSDN file? I didn't see any attribution to it in the source.